bricks@r0.1: stencils/3axis_cl.cpp:11:10: fatal error: CL/opencl.hpp: No such file or directory

### Steps to reproduce the issue

`bricks@r0.1` is failing to build using:
* `spack@develop` (a3354a547fe6c9bb23c49ab610970a01f9081719 from `Sun Apr 23 08:27:26 2023 -0500`)
* Ubuntu 20.04, X86_64
* GCC 11.2.0

Concrete spec: [bricks.spec.json.txt](https://github.com/spack/spack/files/11315072/bricks.spec.json.txt)

<details><summary>Concretization:</summary>
<pre>
 -   bricks@r0.1%gcc@11.2.0~cuda~ipo build_system=cmake build_type=RelWithDebInfo generator=make arch=linux-ubuntu20.04-x86_64
[+]      ^autoconf@2.69%gcc@11.2.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-ubuntu20.04-x86_64
[+]          ^m4@1.4.19%gcc@11.2.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-x86_64
[+]              ^diffutils@3.9%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]              ^libsigsegv@2.14%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]          ^perl@5.36.0%gcc@11.2.0+cpanm+open+shared+threads build_system=generic arch=linux-ubuntu20.04-x86_64
[+]              ^berkeley-db@18.1.40%gcc@11.2.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu20.04-x86_64
[+]              ^bzip2@1.0.8%gcc@11.2.0~debug~pic+shared build_system=generic arch=linux-ubuntu20.04-x86_64
[+]              ^gdbm@1.23%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                  ^readline@8.2%gcc@11.2.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu20.04-x86_64
[+]              ^zlib@1.2.13%gcc@11.2.0+optimize+pic+shared build_system=makefile arch=linux-ubuntu20.04-x86_64
[+]      ^automake@1.16.5%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]      ^cmake@3.26.3%gcc@11.2.0~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-ubuntu20.04-x86_64
[+]          ^ncurses@6.3%gcc@11.2.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]          ^openssl@1.1.1t%gcc@11.2.0~docs~shared build_system=generic certs=mozilla arch=linux-ubuntu20.04-x86_64
[+]              ^ca-certificates-mozilla@2023-01-10%gcc@11.2.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]      ^gmake@4.4.1%gcc@11.2.0~guile build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]      ^libtool@2.4.7%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]      ^mpich@4.1%gcc@11.2.0~argobots~cuda+fortran~hwloc+hydra+libxml2+pci~rocm+romio~slurm~two_level_namespace~vci~verbs~wrapperrpath build_system=autotools datatype-engine=auto device=ch4 netmod=ofi pmi=pmi arch=linux-ubuntu20.04-x86_64
[+]          ^findutils@4.9.0%gcc@11.2.0 build_system=autotools patches=440b954 arch=linux-ubuntu20.04-x86_64
[+]          ^libfabric@1.18.0%gcc@11.2.0~debug~kdreg build_system=autotools fabrics=rxm,sockets,tcp,udp arch=linux-ubuntu20.04-x86_64
[+]          ^libpciaccess@0.17%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]              ^util-macros@1.19.3%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]          ^libxml2@2.10.3%gcc@11.2.0~python build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]              ^libiconv@1.17%gcc@11.2.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]              ^xz@5.4.1%gcc@11.2.0+pic build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]          ^pkgconf@1.8.0%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]          ^yaksa@0.2%gcc@11.2.0~cuda~rocm build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]              ^python@3.7.15%gcc@11.2.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,f2fd060 arch=linux-ubuntu20.04-x86_64
[+]                  ^expat@2.5.0%gcc@11.2.0+libbsd build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                      ^libbsd@0.11.7%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                          ^libmd@1.0.4%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                  ^gettext@0.21.1%gcc@11.2.0+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                      ^tar@1.34%gcc@11.2.0 build_system=autotools zip=pigz arch=linux-ubuntu20.04-x86_64
[+]                          ^pigz@2.7%gcc@11.2.0 build_system=makefile arch=linux-ubuntu20.04-x86_64
[+]                          ^zstd@1.5.5%gcc@11.2.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]                  ^libffi@3.4.4%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                  ^libxcrypt@4.4.33%gcc@11.2.0~obsolete_api build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                  ^sqlite@3.40.1%gcc@11.2.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]                  ^util-linux-uuid@2.38.1%gcc@11.2.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
</pre></details>


Error:
```
$> spack install -f ./bricks.spec.json
...
==> Installing bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344
==> No binary for bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344 found: installing from source
==> No patches needed for bricks
==> bricks: Executing phase: 'cmake'
==> bricks: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/gmake-4.4.1-bdi4lzxyoq3s53nmmnnmd6xkedbl5mt7/bin/make' '-j48'

4 errors found in build log:
     168    [ 54%] Building CXX object single/CMakeFiles/single-cpu.dir/cpu.cpp.o
     169    cd /tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-upj52av/single && /spack/lib/spack/env/gcc/g++  -I/tmp/root/spack-stage/spack-stage-b
            ricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-src/include -I/tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-src -I/spack/opt/spack/linux-u
            buntu20.04-x86_64/gcc-11.2.0/mpich-4.1-bohziz4ihuyn2622dgazarjeqbpfokdv/include -mavx2 -fopenmp -O2 -g -DNDEBUG -std=c++14 -march=native -MD -MT single/CMakeFiles/single-cpu.dir/c
            pu.cpp.o -MF CMakeFiles/single-cpu.dir/cpu.cpp.o.d -o CMakeFiles/single-cpu.dir/cpu.cpp.o -c /tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-s
            rc/single/cpu.cpp
     170    [ 57%] Building CXX object single/CMakeFiles/single-cpu.dir/__/stencils/3axis-out.cpp.o
     171    [ 60%] Building CXX object single/CMakeFiles/single-cpu.dir/__/stencils/copy.cpp.o
     172    cd /tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-upj52av/single && /spack/lib/spack/env/gcc/g++  -I/tmp/root/spack-stage/spack-stage-b
            ricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-src/include -I/tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-src -I/spack/opt/spack/linux-u
            buntu20.04-x86_64/gcc-11.2.0/mpich-4.1-bohziz4ihuyn2622dgazarjeqbpfokdv/include -mavx2 -fopenmp -O2 -g -DNDEBUG -std=c++14 -march=native -MD -MT single/CMakeFiles/single-cpu.dir/_
            _/stencils/3axis-out.cpp.o -MF CMakeFiles/single-cpu.dir/__/stencils/3axis-out.cpp.o.d -o CMakeFiles/single-cpu.dir/__/stencils/3axis-out.cpp.o -c /tmp/root/spack-stage/spack-stag
            e-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-src/stencils/3axis-out.cpp
     173    cd /tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-upj52av/single && /spack/lib/spack/env/gcc/g++  -I/tmp/root/spack-stage/spack-stage-b
            ricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-src/include -I/tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-src -I/spack/opt/spack/linux-u
            buntu20.04-x86_64/gcc-11.2.0/mpich-4.1-bohziz4ihuyn2622dgazarjeqbpfokdv/include -mavx2 -fopenmp -O2 -g -DNDEBUG -std=c++14 -march=native -MD -MT single/CMakeFiles/single-cpu.dir/_
            _/stencils/copy.cpp.o -MF CMakeFiles/single-cpu.dir/__/stencils/copy.cpp.o.d -o CMakeFiles/single-cpu.dir/__/stencils/copy.cpp.o -c /tmp/root/spack-stage/spack-stage-bricks-r0.1-u
            pj52av6va2xhx5fedmaqeio63eu4344/spack-src/stencils/copy.cpp
  >> 174    /tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-src/stencils/3axis_cl.cpp:11:10: fatal error: CL/opencl.hpp: No such file or directory
     175       11 | #include <CL/opencl.hpp>
     176          |          ^~~~~~~~~~~~~~~
     177    compilation terminated.
  >> 178    make[2]: *** [single/CMakeFiles/single-opencl.dir/build.make:93: single/CMakeFiles/single-opencl.dir/__/stencils/3axis_cl.cpp.o] Error 1
     179    make[2]: *** Waiting for unfinished jobs....
     180
     181    /tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-src/weak
     182    main.cpp 34
     183    Total of 1 fixed
     184    A total of 1 stages

     ...

     229    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-upj52av'
     230    /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/gmake-4.4.1-bdi4lzxyoq3s53nmmnnmd6xkedbl5mt7/bin/make  -f weak/CMakeFiles/weak-onesided.dir/build.make weak/CMakeFiles/weak-on
            esided.dir/build
     231    make[2]: Entering directory '/tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-upj52av'
     232    [ 80%] Building CXX object weak/CMakeFiles/weak-onesided.dir/onesided-out.cpp.o
     233    cd /tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-upj52av/weak && /spack/lib/spack/env/gcc/g++  -I/tmp/root/spack-stage/spack-stage-bri
            cks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-src/include -I/tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-src -I/spack/opt/spack/linux-ubu
            ntu20.04-x86_64/gcc-11.2.0/mpich-4.1-bohziz4ihuyn2622dgazarjeqbpfokdv/include -mavx2 -fopenmp -O2 -g -DNDEBUG -std=c++14 -march=native -MD -MT weak/CMakeFiles/weak-onesided.dir/on
            esided-out.cpp.o -MF CMakeFiles/weak-onesided.dir/onesided-out.cpp.o.d -o CMakeFiles/weak-onesided.dir/onesided-out.cpp.o -c /tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6
            va2xhx5fedmaqeio63eu4344/spack-src/weak/onesided-out.cpp
     234    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-upj52av'
  >> 235    make[1]: *** [CMakeFiles/Makefile2:302: single/CMakeFiles/single-opencl.dir/all] Error 2
     236    make[1]: *** Waiting for unfinished jobs....
     237    [ 82%] Linking CXX executable mpi
     238    cd /tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-upj52av/single && /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/cmake-3.26.3-q
            jlpperx6scndo5524d5l2brpho3h5sm/bin/cmake -E cmake_link_script CMakeFiles/single-mpi.dir/link.txt --verbose=1
     239    /spack/lib/spack/env/gcc/g++ -mavx2 -fopenmp -O2 -g -DNDEBUG "CMakeFiles/single-mpi.dir/mpi.cpp.o" "CMakeFiles/single-mpi.dir/__/src/brick-mpi.cpp.o" -o mpi  -Wl,-rpath,/spack/opt
            /spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/mpich-4.1-bohziz4ihuyn2622dgazarjeqbpfokdv/lib ../src/libbrick.a ../src/libbrickhelper.a /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-1
            1.2.0/mpich-4.1-bohziz4ihuyn2622dgazarjeqbpfokdv/lib/libmpicxx.so /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/mpich-4.1-bohziz4ihuyn2622dgazarjeqbpfokdv/lib/libmpi.so
     240    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-upj52av'
     241    [ 82%] Built target single-mpi

     ...

     262    [ 97%] Linking CXX executable cpu-4d
     263    cd /tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-upj52av/weak && /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/cmake-3.26.3-qjl
            pperx6scndo5524d5l2brpho3h5sm/bin/cmake -E cmake_link_script CMakeFiles/weak-cpu-4d.dir/link.txt --verbose=1
     264    /spack/lib/spack/env/gcc/g++ -mavx2 -fopenmp -O2 -g -DNDEBUG "CMakeFiles/weak-cpu-4d.dir/main-4d-out.cpp.o" "CMakeFiles/weak-cpu-4d.dir/args.cpp.o" -o cpu-4d  -Wl,-rpath,/spack/op
            t/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/mpich-4.1-bohziz4ihuyn2622dgazarjeqbpfokdv/lib ../src/libbrick.a ../src/libbrick-mpi.a ../src/libbrickhelper.a /spack/opt/spack/linux-u
            buntu20.04-x86_64/gcc-11.2.0/mpich-4.1-bohziz4ihuyn2622dgazarjeqbpfokdv/lib/libmpicxx.so /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.2.0/mpich-4.1-bohziz4ihuyn2622dgazarjeqbp
            fokdv/lib/libmpi.so /usr/lib/x86_64-linux-gnu/librt.so
     265    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-upj52av'
     266    [ 97%] Built target weak-cpu-4d
     267    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-upj52av'
  >> 268    make: *** [Makefile:139: all] Error 2

See build log for details:
  /tmp/root/spack-stage/spack-stage-bricks-r0.1-upj52av6va2xhx5fedmaqeio63eu4344/spack-build-out.txt
```

### Error message

<details><summary>Error message</summary>

<pre>
...
</pre></details>


### Information on your system

* **Spack:** 0.20.0.dev0 (a3354a547fe6c9bb23c49ab610970a01f9081719)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[bricks.spec.json.txt](https://github.com/spack/spack/files/11315068/bricks.spec.json.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/11315069/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/11315070/spack-build-out.txt)

@drhansj @ztuowen

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/bricks/bricks-cmakelists-option-opencl.patch
+++ b/var/spack/repos/builtin/packages/bricks/bricks-cmakelists-option-opencl.patch
@@ -1,0 +1,17 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 58dcbd4..f0658eb 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -43,7 +43,11 @@ endif()
+ 
+ find_package(OpenMP REQUIRED)
+ find_package(MPI)
+-find_package(OpenCL 2.0)
++
++option(BRICK_USE_OPENCL "Use OpenCL targets" ON)
++if (BRICK_USE_OPENCL)
++    find_package(OpenCL 2.0)
++endif()
+ 
+ option(BRICK_USE_MEMFD "Using memfd instead of shm_open, supported on Linux >= 3.17 with \"CONFIG_MEMFD_CREATE\"" OFF)
+ if (BRICK_USE_MEMFD)

--- a/var/spack/repos/builtin/packages/bricks/package.py
+++ b/var/spack/repos/builtin/packages/bricks/package.py
@@ -35,10 +35,13 @@ class Bricks(CMakePackage):
     depends_on("cuda", when="+cuda")
     depends_on("mpi")
 
+    patch("bricks-cmakelists-option-opencl.patch")
+
     def cmake_args(self):
         """CMake arguments for configure stage"""
-        args = []
-
+        args = [self.define_from_variant("BRICK_USE_OPENCL", "cuda")]
+        if "+cuda" in self.spec:
+            args.append(f"-DOCL_ROOT:STRING={self.spec['opencl-clhpp'].prefix}")
         return args
 
     def flag_handler(self, name, flags):

